pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -406,22 +397,26 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun, get
 		message := fmt.Sprintf("PipelineRun %s/%s awaiting remote resource", pr.Namespace, pr.Name)
 		pr.Status.MarkRunning(ReasonResolvingPipelineRef, message)
 		return nil
-	case errors.Is(err, trustedresources.ErrResourceVerificationFailed):
-		message := fmt.Sprintf("PipelineRun %s/%s referred pipeline failed signature verification", pr.Namespace, pr.Name)
-		pr.Status.MarkFailed(ReasonResourceVerificationFailed, message)
-		return controller.NewPermanentError(err)
 	case err != nil:
 		logger.Errorf("Failed to determine Pipeline spec to use for pipelinerun %s: %v", pr.Name, err)
 		pr.Status.MarkFailed(ReasonCouldntGetPipeline,
 			"Error retrieving pipeline for pipelinerun %s/%s: %s",
 			pr.Namespace, pr.Name, err)
 		return controller.NewPermanentError(err)
-	default:
-		// Store the fetched PipelineSpec on the PipelineRun for auditing
-		if err := storePipelineSpecAndMergeMeta(ctx, pr, pipelineSpec, pipelineMeta); err != nil {
-			logger.Errorf("Failed to store PipelineSpec on PipelineRun.Status for pipelinerun %s: %v", pr.Name, err)
+	}
+
+	// Skip the verification if it is not inline pipeline;
+	if pr.Status.PipelineSpec == nil {
+		if err := c.verifyPipeline(ctx, pr, pipelineMeta, pipelineSpec); err != nil {
+			return err
 		}
 	}
+	// SetDefaults after the verification
+	pipelineSpec.SetDefaults(ctx)
+	// Store the fetched PipelineSpec on the PipelineRun for auditing
+	if err := storePipelineSpecAndMergeMeta(ctx, pr, pipelineSpec, pipelineMeta); err != nil {
```

Comment:
Those need to be called after verification

Reply:
Can you add more details in the comments why this needs to be done in this order?

Reply:
sure!